FrontEnd/src/main/java/resourceDisplay/IndexController.java

```diff
@@ -46,29 +46,41 @@ public String index() {
     }
 
     /**
-     * This page returns all the assets in the DB
+     * This page returns all assets in the DB with different filtering options
      *
-     * @param principal - Used to check authentication
-     * @param model     - Used to show table to user
-     * @return the allassets template
+     * @param principal    - Used to check authentication
+     * @param model        - Used to show table to user
+     * @param filterObject - Used to get filters from the user
+     * @return the bystatus template
      */
     @GetMapping("/allassets")
-    public String getAll(@AuthenticationPrincipal OAuth2User principal, Model model) {
+    public String test(@AuthenticationPrincipal OAuth2User principal, Model model,
+                       @ModelAttribute FilterObject filterObject) {
         SpannerOptions options =
                 SpannerOptions.newBuilder().setProjectId(SPANNER_PROJECT_ID).build();
         spanner = options.getService();
         db = DatabaseId.of(SPANNER_PROJECT_ID, SPANNER_INSTANCE_ID, SPANNER_DATABASE_ID);
         dbClient = spanner.getDatabaseClient(db);
         AssetsRepository assets = new AssetsRepository();
-        //TODO use workspaceID from user/workspace/project
-        ResultListObject resultListObject = assets.getAllAssets(dbClient);
-        model.addAttribute("displayNames", resultListObject.columnDisplays);
-        model.addAttribute("allAssets", resultListObject.columnResults);
+        model.addAttribute("filterObject", filterObject);
+        //TODO use real workspaceID from user/workspace/project table this is temporary (using "noasan")
```

Comment:
any idea when are we planning to move ahead with this TODO?

Reply:
Will work on this after we finish creating the service account table (working on it now).